src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ReferralServiceTest.kt

```diff
@@ -1,57 +1,111 @@
     )
 
-    val updated = ReferralService.updateReferral(newOriginal(), update)
-    assertThat(updated.id).isEqualTo(UUID.fromString("3529d30e-3480-4686-b14b-485beb3431ed"))
-    assertThat(updated.created).isEqualTo(OffsetDateTime.parse("2021-05-05T13:00:01+01:00"))
+    val updated = referralService.updateDraftReferral(referral.id!!, draftReferral)
+    assertThat(updated!!.id).isEqualTo(referral.id!!)
+    assertThat(updated.created).isEqualTo(referral.created)
   }
 
   @Test
   fun `null fields in the update do not overwrite original fields`() {
-    val update = DraftReferral(completionDeadline = null)
+    val referral = Referral(completionDeadline = LocalDate.of(2021, 6, 26))
+    entityManager.persist(referral)
+    entityManager.flush()
+
+    val draftReferral = DraftReferral(completionDeadline = null)
 
-    val updated = ReferralService.updateReferral(newOriginal(), update)
-    assertThat(updated.completionDeadline).isEqualTo(LocalDate.parse("2021-06-26"))
+    val updated = referralService.updateDraftReferral(referral.id!!, draftReferral)
+    assertThat(updated!!.completionDeadline).isEqualTo(LocalDate.of(2021, 6, 26))
   }
 
   @Test
   fun `non-null fields in the update overwrite original fields`() {
-    val update = DraftReferral(completionDeadline = LocalDate.parse("2020-12-01"))
+    val referral = Referral(completionDeadline = LocalDate.of(2021, 6, 26))
+    entityManager.persist(referral)
+    entityManager.flush()
 
-    val updated = ReferralService.updateReferral(newOriginal(), update)
-    assertThat(updated.completionDeadline).isEqualTo(LocalDate.parse("2020-12-01"))
+    val draftReferral = DraftReferral(completionDeadline = LocalDate.of(2020, 12, 1))
+
+    val updated = referralService.updateDraftReferral(referral.id!!, draftReferral)
+    assertThat(updated!!.completionDeadline).isEqualTo(LocalDate.of(2020, 12, 1))
   }
 
   @Test
   fun `update mutates the original object`() {
-    val original = newOriginal()
-    val update = DraftReferral(completionDeadline = LocalDate.parse("2020-12-01"))
+    val referral = Referral(completionDeadline = LocalDate.of(2021, 6, 26))
+    entityManager.persist(referral)
+    entityManager.flush()
+
+    val draftReferral = DraftReferral(completionDeadline = LocalDate.of(2020, 12, 1))
+
+    val updated = referralService.updateDraftReferral(referral.id!!, draftReferral)
+    assertThat(updated!!.completionDeadline).isEqualTo(LocalDate.of(2020, 12, 1))
+  }
+
+  @Test
+  fun `update successfully persists the updated draft referral`() {
+    val referral = Referral(completionDeadline = LocalDate.of(2021, 6, 26))
+    entityManager.persist(referral)
+    entityManager.flush()
+
+    val draftReferral = DraftReferral(completionDeadline = LocalDate.of(2020, 12, 1))
+
+    val updated = referralService.updateDraftReferral(referral.id!!, draftReferral)
+    assertThat(updated!!.id).isEqualTo(referral.id)
```

Comment:
in this test we aren't actually testing that the update has been saved to the database. i think we need to make a call to `getDraftReferral` to check.

Reply:
Removed check against returned update to persisted object from getDraftReferral